src/components/userFeedback/UserFeedback.spec.tsx

```
@@ -0,0 +1,143 @@
+
+
+        testText = 'hello this is a test over here';
+        wrongText = 'hello this is the wrong text over here';
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).not.toBe(wrongText);
+      });
+    });
+
+    describe('style for each state', () => {
+      describe('non existent state', () => {
+        it('should be invisible if state provided does not exist', () => {
+          let nonExistentState = 'NON_EXISTENT_STATE';
+          let emptyState = '';
+
+          expect(getShallowOutput(generateProps('', nonExistentState, '')).hasClass('hidden')).toBe(true);
+          expect(getShallowOutput(generateProps('', emptyState, '')).hasClass('hidden')).toBe(true);
+        });
+      });
+
+      describe('VALID state', () => {
+        it('should be invisible on state VALID without displayOnShow prop', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '')).hasClass('hidden')).toBe(true);
+        });
+
+        it('should be invisible on state VALID, even with prop displayOnShow provided', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '')).hasClass('hidden')).toBe(true);
+        });
+      });
+
+      describe('WARNING state', () => {
+        it('should be visible (without prop displayOnShow)', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, '')).hasClass('block')).toBe(true);
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, '')).hasClass('hidden')).toBe(false);
+        });
+
+        it('the display value should be the value of the displayOnShow prop', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, 'inline-block')).hasClass('inline-block')).toBe(true);
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, 'inline-block')).hasClass('hidden')).toBe(false);
```

I think it should behave naturally and hide the class, we could add a test for that. Otherwise, could we throw an error with TypeScript at compile time in case the class is set as hidden, or is simply not a displayClass?